### PR TITLE
Add file extension when using `absoluteRuntime`

### LINKED ISSUE #12824: Webpack 5 error for missing extention for regenerator runtime when `absoluteRuntime` is used and a .mjs file is transformed
## Bug Report

- [x] I would like to work on a fix!

**Current behavior**

babel-loader is used with webpack with the `"@babel/plugin-transform-runtime` plugin and `regenerator` AND `absoluteRuntime` path . Weback 5 (since webpack v5.0.0-beta.30,) when parses an es module (.mjs or type: module), tries to enforce the node spec regarding the deep imports, where the file extention should be present and when it's missing, it throws an error:
```
ERROR in ./src/index.mjs 1:0-114
Module not found: Error: Can't resolve '.../node_modules/@babel/runtime/regenerator' in '...\src'
Did you mean 'index.js'?
BREAKING CHANGE: The request '.../node_modules/@babel/runtime/regenerator' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
```

- [REPL]
minimal reproducible repo: https://github.com/mbehzad/webpack-babel-absoluteruntime-issue

**Input Code**
cmd: `webpack build`    

**Expected behavior**

When the runtime imports are injected, and the user has specifies a path to babel-runtime and therefor not the module name (and package exports maps) are used, the file extensions should be in the imports so webpack is happy.

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
- Filename: `weback.config.js`

```js
const path = require("path");

module.exports = {
  mode: "development",
  entry: "./src/index.mjs",
  module: {
    rules: [
      {
        test: /\.m?js$/,
        use: {
          loader: "babel-loader",
          options: {
            presets: [
              [
                require.resolve("@babel/preset-env"),
                {
                  targets: {
                    ie: 11
                  }
                }
              ]
            ],
            plugins: [
              [
                require.resolve("@babel/plugin-transform-runtime"),
                {
                  corejs: false,
                  regenerator: true,
                  useESModules: false,
                  helpers: false,
                  absoluteRuntime: path.dirname(
                    require.resolve("@babel/runtime/package.json")
                  )
                }
              ]
            ]
          }
        }
      }
    ]
  }
};

```

**Environment**

```
- Babel version(s): [v7.12.0]
- Node/npm version: [Node 15.3.0/npm 7.0.14]
- OS: [Windows 10]
- Monorepo: [no]
- How you are using Babel: [ `webpack`]
- {
    "@babel/core": "7.2.0",
    "@babel/plugin-transform-runtime": "7.12.15",
    "@babel/preset-env": "7.12.16",
    "babel-loader": "8.2.2",
    "webpack": "5.22.0",
    "webpack-cli": "4.5.0"
  }
  ```

**Possible Solution**

I guess in `babel-plugin-transform-runtime/index.js` the extentions / index.js should be added to the end of the runtime module names for the imports.
**Additional context**

N/A

### PR DESCRIPTION
**Pull request by mbehzad**
es module imports need the file extension (e.g. `import "@babel/runtime/helpers/jsx.js"`, Or the filenames being listed in the package.json's subpath exports (e.g. `import "@babel/runtime/helpers/jsx"` + pkg: `"./helpers/jsx": "./helpers/jsx.js"`). when the user passes a path via `absoluteRuntime` then the rendered require statement is not the module name + subpath which will be resolved via pkg.json but rather the absolute path to the file. for this case, add the file extension / index.js to prevent bundlers from raising a warning.

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #12824  
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes (modified existing ones)
| Documentation PR Link    |  No
| Any Dependency Changes?  | No
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/12827"></a>

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
@mbehzad Sorry for the long delay before reviewing this PR: there have been a lot of `@babel/runtime` changes between 7.13.0 and 7.13.7 so we have been waiting to review this one.

Could you rebase this PR, and add a test with `absoluteRuntime: true` + `useESModules: true`? :pray:

**Comment by mbehzad** [comment]
Hi @nicolo-ribaudo , I've added my changes to the current code, and added a test for `absoluteRuntime` + `useESModules` option.
**But** the regenerator import statement injection seems to have been moved to the `babel-plugin-polyfill-regenerator` module. It needs to have the `/index.js` at the end for the `absoluteRuntime` case. I'm wondering if i should add similar to the `ext` property in the runtime compat option, a flag/property to add that. or even use the `ext` prop for this and call `/index.js` the "extension" here. Let me know what is preferred. Thanx.

**Comment by charlessuh** [comment]
@mbehzad Hi, I'm just another random person who also ran into this issue, but could you rebase this?

I'm wondering if perhaps the regenerator issue can be resolved later as a separate PR, as it may require a change to a package that lives in another repo (https://github.com/babel/babel-polyfills)

**Comment by nicolo-ribaudo** [comment]
I have rebased https://github.com/babel/babel-polyfills/pull/79. As soon as it's merged, I'll rebase this one.

File: packages/babel-plugin-transform-runtime/test/fixtures/absoluteRuntime/relative/subfolder/node_modules/@babel/runtime/helpers/classCallCheck.js [hunk 1]
```diff
@@ -0,0 +1 @@
+// empty
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-transform-runtime/test/fixtures/absoluteRuntime/relative/subfolder/node_modules/@babel/runtime/helpers/classCallCheck.js:1]: This empty file was needed to be able to `require.resolve` it.

**Review by JLHwung — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**